specutils/analysis/flux.py

```diff
@@ -121,6 +129,15 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None):
 
     line_flux = _compute_line_flux(spectrum, regions)
 
+    # Account for the existence of a mask. A correction factor
+    # is applied to the continuum value to account for the
+    # data points that are not included in the line flux due to
+    # masking.
+    if hasattr(spectrum, 'mask') and spectrum.mask is not None:
+        valid = np.sum(np.where(calc_spectrum.mask, 0., 1.))
+        factor = valid / len(spectral_axis)
```

Comment:
Since `.mask` is a boolean mask, something like

```
factor = (~calc_spectrum.mask).sum() / len(spectral_axis)
```

can be a little clearer and more efficient.